BitSliceIndexing/bsi.go

```diff
@@ -380,19 +472,35 @@ func (b *BSI) UnmarshalBinary(bitData [][]byte) error {
 			continue
 		}
 		if b.BitCount() < i {
-			b.bA = append(b.bA, roaring.NewBitmap())
+			newBm := roaring.NewBitmap()
+			if b.runOptimized {
+				newBm.RunOptimize()
+			}
+			b.bA = append(b.bA, newBm)
 		}
 		if err := b.bA[i-1].UnmarshalBinary(bitData[i]); err != nil {
 			return err
+		} else {
```

Comment:
This else is unnecessary. Rest of the code should not be in braces.

Reply:
Done